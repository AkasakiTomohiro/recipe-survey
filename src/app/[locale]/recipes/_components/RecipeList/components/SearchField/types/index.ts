/**
 * @file index.ts
 * @author Artan's Projects
 * @copyright © 2024 Artan's Projects. All rights reserved.
 */

export interface ISearchFieldProps {

  /**
   * 検索ワード
   */
  searchWord: string;

  /**
   * 検索関数
   */
  searchFilter: (word: string) => void;
}