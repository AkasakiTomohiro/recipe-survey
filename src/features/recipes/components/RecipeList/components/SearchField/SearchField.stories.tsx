/**
 * @file SearchField.stories.tsx
 * @author Artan's Projects
 * @copyright © 2024 Artan's Projects. All rights reserved.
 */

import type { Meta, StoryObj } from '@storybook/react';

import { SearchField } from './SearchField';

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
const meta = {
  title     : 'features/Recipe/RecipeList/SearchField',
  component : SearchField,
  parameters: {
  },

  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ['autodocs']
} satisfies Meta<typeof SearchField>;

export default meta;
type Story = StoryObj<typeof meta>;

// More on writing stories with args: https://storybook.js.org/docs/writing-stories/args
export const Primary: Story = {
  args: {
    searchWord: 'バッテリー',
    searchFilter: (word: string) => {
      console.log(word);
    }
  }
};
