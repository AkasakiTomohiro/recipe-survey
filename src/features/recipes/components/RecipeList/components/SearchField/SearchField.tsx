/**
 * @file SearchField.tsx
 * @author Artan's Projects
 * @copyright © 2024 Artan's Projects. All rights reserved.
 */

import { useCallback } from 'react';
import { SearchIcon } from '~/components/Icon';
import { useTransLocations } from '~/i18n';

import { InputAdornment, TextField } from '@mui/material';

import { ISearchFieldProps } from './types';

export function SearchField(props: ISearchFieldProps): JSX.Element {
  const t = useTransLocations();

  /**
   * 検索する関数
   */
  const handleSearch = useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
    props.searchFilter(event.target.value);
  }, [props]);

  return (
    <TextField
      label={t('common.search')}
      InputProps={{
        endAdornment: (
          <InputAdornment position="end">
            <SearchIcon />
          </InputAdornment>
        )
      }}
      fullWidth
      variant="standard"
      value={props.searchWord}
      onChange={handleSearch}
      margin="normal"
    />
  );
}