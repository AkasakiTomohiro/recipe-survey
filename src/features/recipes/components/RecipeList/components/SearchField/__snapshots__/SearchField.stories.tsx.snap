// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeSurvey/recipes/components/SearchField Primary smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root">
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard css-zx681a-MuiFormLabel-root-MuiInputLabel-root"
         data-shrink="true"
         for=":r0:"
         id=":r0:-label"
  >
    検索
  </label>
  <div class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-se4mu1-MuiInputBase-root-MuiInput-root">
    <input aria-invalid="false"
           id=":r0:"
           class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
           type="text"
           value="バッテリー"
    >
    <div class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root">
      <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
           focusable="false"
           aria-hidden="true"
           viewbox="0 0 24 24"
           data-testid="SearchIcon"
      >
        <path d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14">
        </path>
      </svg>
    </div>
  </div>
</div>
`;
